paddle/pybind/pybind.cc

```diff
@@ -25,9 +26,143 @@ namespace pd = paddle::framework;
 
 USE_OP(add_two);
 
+struct PlaceDebugString : public boost::static_visitor<std::string> {
+  std::string operator()(const paddle::platform::GPUPlace& place) const {
+    return "GPU(" + std::to_string(place.device) + ")";
+  }
+
+  std::string operator()(const paddle::platform::CPUPlace& place) const {
+    return "CPU";
+  }
+};
+
+template <typename T>
```

Comment:
@reyoung I don't understand what the issue that C++11 doesn't have static if.  I built the following test program using GCC with `-std=c++11`

```cpp
#include <iostream>
#include <tuple>

template <size_t TypeIndex, typename... ElementTypes>
void SelectType() {
  using Types = std::tuple<ElementTypes...>;

  if (TypeIndex >= std::tuple_size<Types>::value) {
    std::cout << "Larger\n";
  } else {
    std::cout << "OK\n";
  }
}
     
int main() {
  SelectType<0, float, int>();
  SelectType<1, float, int>();
  SelectType<2, float, int>();
  return 0;
}
```

and it works:

```bash
$ g++ a.cc -std=c++11 -o a && ./a
OK
OK
Larger
```